File: message.go
package main

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

// Message ...
type Message struct {
	ID          primitive.ObjectID `bson:"_id, omitempty"`
	Message     string             `bson:"message" json:"message"`
	MessageFrom User               `bson:"messageFrom" json:"messageFrom"`
	MessageTo   User               `bson:"messageTo" json:"messageTo"`
	SentAt      time.Time          `bson:"sentAt" json:"sentAt"`
}

File: src/main/java/com/friends/message/Message/Message.java
package com.friends.message.Message;

import java.util.Date;
import javax.validation.constraints.NotNull;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.Data;
import com.friends.message.User.User;


@Data
@Document
public class Message {

    @Id
    private String messageId;
    
    @NotNull
    private String message; 
    @NotNull
    private User messageFrom;
    @NotNull
    private User messageTo;

    @NotNull
    private Date sentAt;

    Message(String message, User messageFrom, User messageTo) {

        this.message = message; 
        this.messageFrom = messageFrom; 
        this.messageTo = messageTo; 
        Date date = new Date();
        sentAt= new Date(date.getTime()); 
    }

    Message() {}
}